ZkLobbyServer/DiscordRelaySource.cs

```diff
@@ -50,13 +55,25 @@ public void SetTopic(string channel, string topic)
             }
         }
 
-
         public void SendMessage(ChatRelayMessage m)
         {
             try
             {
                 if (m.Source != source)
                 {
+                    //Translate mentions of nicknames to discord mentions
+                    var userIdsByNickname = discord.GetGuild(serverID).Users.ToDictionary(x => x.Nickname, x => x.Id.ToString(), StringComparer.OrdinalIgnoreCase);
+                    m.Message = Regex.Replace(m.Message, "(\\w+)", match => userIdsByNickname.ContainsKey(match.Groups[1].Value) ? 
+                            string.Format("<@{0}>", userIdsByNickname[match.Groups[1].Value]) : match.Groups[1].Value);
+
+                    //Block any mentions of an entire role via ID
+                    var roleIds = discord.GetGuild(serverID).Roles.Select(x => x.Id.ToString()).ToList();
+                    m.Message = ReplaceMention(m.Message, match => roleIds.Contains(match.Groups[1].Value) ? "" : match.Groups[1].Value);
+
+                    //Block any mentions of an entire role via Name
+                    var roleNames = discord.GetGuild(serverID).Roles.Select(x => x.Name).ToList();
+                    roleNames.ForEach(role => m.Message = m.Message.Replace(string.Format("@{0}", role), ""));
```

Comment:
How about
```suggestion
                    roleNames.ForEach(role => m.Message = m.Message.Replace(string.Format("@{0}", role), role));
```
For stuff like

> is any @admin online?

Reply:
I thought about that, but what about `@@Admin`?

Reply:
Replace with `string.Format(" {0}", role)` or somesuch then?